### PR TITLE
InOb polyfill trigger new added element w/o tick

### LINKED ISSUE #7992: InOb polyfill: should trigger automatically for new elements w/o ticks
See this unit test to repro: #7991.

Per spec, the last step for `IntersectionObserver.observe()` API is:
> Schedule an iteration of the event loop in the root's browsing context.

Further, I believe `tick` calls need to be **debounced**. In many cases `scroll` event is called way to rapidly to call measures immediately.

### PR DESCRIPTION
**Pull request by zhouyx**
fix #7992 
- Always store the last tick viewport position, and use it to calculate change entry to new added element w/o tick
- If last tick viewport position is not available, do nothing.
- IntersectionObserverPolyfill creator is responsible for the very first tick during initialization.

### CODE REVIEW

File: src/intersection-observer-polyfill.js [hunk 1]
```diff
@@ -201,6 +202,12 @@ export class IntersectionObserverPolyfill {
         this.threshold_[this.threshold_.length - 1] <= 1,
         'Threshold should be in the range from "[0, 1]"');
 
+    /** @private {?./layout-rect.LayoutRectDef} */
+    this.lastViewportRect_ = null;
+
+    /** @private {./layout-rect.LayoutRectDef|undefined} */
+    this.opt_lastIframeRect_ = undefined;
```
Comment (dvoytenko) [inline src/intersection-observer-polyfill.js]: Instance/class vars cannot have `opt_`  prefix. Please remove.

**Comment by dvoytenko** [comment]
LGTM with one request.

File: src/intersection-observer-polyfill.js (hunk 1 above)
Comment (zhouyx) [inline src/intersection-observer-polyfill.js]: removed

File: src/intersection-observer-polyfill.js [hunk 2]
```diff
@@ -227,11 +234,22 @@ export class IntersectionObserverPolyfill {
       }
     }
 
-    // push new observed element
-    this.observeEntries_.push({
+    const newEntry = {
```
Comment (lannka) [inline src/intersection-observer-polyfill.js]: as discussed, please avoid overloading the name `entry` here. Can be called `state` instead.

Comment (zhouyx) [inline src/intersection-observer-polyfill.js]: done

File: extensions/amp-analytics/0.1/visibility-impl.js [hunk 3]
```diff
@@ -300,6 +300,7 @@ export class Visibility {
             onIntersectionChanges, {threshold: DEFAULT_THRESHOLD});
         //TODO: eventually this is go into the proposed layoutManager.
         const viewport = viewportForDoc(this.ampdoc);
+        this.intersectionObserver_.tick(viewport.getRect());
```
Comment (jridgewell) [inline extensions/amp-analytics/0.1/visibility-impl.js]: Nit: call `ticker` instead of duplicating its code.

**Pull request merged**